app/src/main/java/org/mozilla/focus/utils/Settings.kt

```diff
@@ -49,6 +49,15 @@ class Settings private constructor(context: Context) {
             getPreferenceKey(R.string.pref_key_performance_block_webfonts),
             false)
 
+    fun shouldRequestDesktopSite(): Boolean =
+            preferences.getBoolean(
+                getPreferenceKey(R.string.pref_key_performance_request_desktop_site),
+                false)
+
+    fun setShouldRequestDesktopSite(shouldRequestDesktopSite: Boolean) =
+            preferences.edit().putBoolean(getPreferenceKey(R.string.pref_key_performance_request_desktop_site),
+                shouldRequestDesktopSite).apply()
+
```

Comment:
What is this setting for? I assume this should be something we just save in the session for this "tab" (like the state of the tracking switch) and no global setting?

Reply:
The specs detailed it not "carrying back" but "carry foward" as a global setting see #822

Reply:
Oh, I missed that part. I'll comment on the issue. I think this is pretty confusing.